sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1275,7 +1275,7 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
           if ($longValue == ($integralType) $longValue) {
             $evPrim = ($integralType) $longValue;
           } else {
-            throw new ArithmeticException("Casting $c to $integralType causes overflow");
+            throw new ArithmeticException("Casting " + $c + " to $integralType causes overflow");
```

Comment:
How about using string interpolation `s" ... $c ..."?

Reply:
They are different. The change you suggested is the same as what it is originally.
Before:
```
Casting columnartorow_value_0 to short causes overflow
```
After:
```
Casting 4567890123456789 to short causes overflow
```

Reply:
Ah, got it. This is in another interpolation `code"""`.   
Sorry...

Reply:
It's OK. I made the same mistake :)